packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -306,29 +306,35 @@ export class XInstaller {
   }
 
   /**
-   * It returns the HTML element to mount the Vue Application. If the `domElement` parameter in the
-   * {@link InstallXOptions} is an Element or a string, then it is used. If it is
-   * not present then a new <div> Element is created and append to the body to be used.
+   * It returns the HTML element to mount the Vue Application. If the `domElement` parameter in
+   * the {@link InstallXOptions} is an Element or an element selector, then this will be used.
+   * The `domElement` can also be a function with the {@link SnippetConfig} as parameter which
+   * returns an Element or element selector to use.
+   * If it is not present, a new <div> Element is created and appended to the body.
    *
-   * @param elementOrSelector - String or Element used to mount the Vue App.
+   * @param domElement - {@link InstallXOptions.domElement | Element, string or function} Used
+   * to mount the Vue Application.
    *
-   * @returns The Element to use as mounting point for the Vue App.
+   * @returns The Element to use as mounting target for the Vue Application.
    * @internal
    */
-  protected getMountingTarget(elementOrSelector?: string | Element): Element {
-    if (typeof elementOrSelector === 'string') {
-      const target = document.querySelector(elementOrSelector);
+  protected getMountingTarget(domElement?: InstallXOptions['domElement']): Element {
+    if (typeof domElement === 'function') {
```

Comment:
It is not necessary and you probably weren't aware of it, but we have an `isFunction` `typeguard` helper 😅 

I don't leave a code suggestion cause you will have to import it anyways, but feel free to do it or take a look at it in case you are curious (reinventing the wheel 💪🏿 ).

Reply:
Overengineering is always in my team 😄 
Changed for the typeguard.